core/jobs/batch_jobs/suggestion_stats_computation_jobs.py

```diff
@@ -623,6 +623,8 @@ def _combine_translation_contribution_stats_objects(
             stat.contribution_dates for stat in stats
         ]
         contribution_dates = contribution_dates.union(*all_contribution_dates)
+        sorted_contribution_dates = sorted(contribution_dates)
+        sorted_contribution_dates_set = set(sorted_contribution_dates)
```

Comment:
Er -- when you make them a set again, they become unordered. I think this sorting is something you'd need to do between the domain object and datastore level (when you convert the domain object to a datastore model).

Reply:
Actually this is the place where I converts the domain object to the model(when running the beam job). Do you want me to update the properties of contribution_dates field in the Model class in gae_models.py file?

Reply:
Sorry, I don't get it. suggestion_registry.TranslationContributionStats is a domain object. At which point do you modify a datastore model?

Reply:
Sorry for the confusion. What we do in this pipeline is:
1. generate stats per contribution
2. combine generated stats per user per language per topic(this is what we are doing here)
3. create stats model from domain object(https://github.com/oppia/oppia/blob/03c870d73b25374fb9342ca75992c29917e1582c/core/jobs/batch_jobs/suggestion_stats_computation_jobs.py#L922)

Basically these domain objects will be directly converted into the model and we don't do any computations while converting the domain object into model.

Reply:
When you create TranslationContributionStatsModel, you are passing translation_contribution_dates as the value of the contribution_dates field. Isn't the former a set and the latter a list?

If so, then you'll need to change the line to something like `contribution_dates=sorted(translation.contribution_dates)` instead, right?

Reply:
Done